include/otf2xx/event/metric.hpp

```diff
@@ -267,12 +576,33 @@ namespace event
             metric_instance_ = mi;
         }
 
+        otf2::definition::metric_class resolve_metric_class() const
+        {
+            return otf2::definition::metric_class{ resolve_metric_class_ref() };
+        }
+
         friend class otf2::writer::local;
 
     private:
-        otf2::definition::detail::weak_ref<otf2::definition::metric_class> metric_class_;
-        otf2::definition::detail::weak_ref<otf2::definition::metric_instance> metric_instance_;
-        std::vector<value_container> values_;
+        template <typename Definition>
+        using weak_ref = otf2::definition::detail::weak_ref<Definition>;
+
+        weak_ref<otf2::definition::metric_class> resolve_metric_class_ref() const
```

Comment:
this shouldn't be called `resolve_metric_class_ref` to prevent confusions with `resolve_metric_class().ref()`